classes/Tools.php

```diff
@@ -3782,17 +3782,25 @@ public static function arrayUnique($array)
     /**
      * Delete unicode class from regular expression patterns.
      *
+     * @deprecated Use PrestaShop\PrestaShop\Core\String\CharacterCleaner::cleanNonUnicodeSupport() instead
+     *
      * @param string $pattern
      *
      * @return string pattern
+     *
+     * @throws Exception
      */
     public static function cleanNonUnicodeSupport($pattern)
     {
-        if (!defined('PREG_BAD_UTF8_OFFSET')) {
-            return $pattern;
+        $context = Context::getContext();
+        $containerFinder = new ContainerFinder($context);
+        $container = $containerFinder->getContainer();
+        if (null === $context->container) {
```

Comment:
is it useful to assign `$context->container` here? there's no need for it in this method
if the aim is to optimize other calls of ContainerFinder, then maybe it should be made in ContainerFinder, although I don't think ti would bring much gain